### PR TITLE
XML Validation script

### PR DESCRIPTION
**Pull request by bhermanmit**
This PR addresses #248. This new script checks two features of OpenMC XML input files:
- Can it be parsed by an XML parser
- Does it match relaxng files that are in the source directory

It can be run with:
```
python xml_validate.py -r {path to relaxng files in OpenMC src} -i {path to input files, optional, assumes current working directory}
```

Play around with it and let me know if I can make it better, it is at least a start. It will not catch my most prevalent mistake which is messing up energy bins in tally filters, but that is because the RNC files say it should be a string and not a list of doubles.

### CODE REVIEW

**Comment by paulromano** [comment]
A suggestion for improving this: you can have our setup.py script install xml_validate.py as well as the RelaxNG schema files. Then xml_validate.py could find the schema files using Python's builting pkgutil module. This would avoid having to specify the source directory.

**Comment by paulromano** [comment]
A short section in the user's manual on this functionality is also warranted.

**Comment by bhermanmit** [comment]
@paulromano - Ok, I installed xml_validate script into bin folder and relaxng files in share folder of install directory using cmake. I also use the xml_validate path when the script is executed to try and search for the relaxng directory. If I can't find it, I ask the user to manually specify it with the -r option. Will get to the user's manual once you are happy with the functionality.

File: src/utils/xml_validate.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+# Get absolute paths
+if options.relaxng is not None:
+    relaxng_path = os.path.abspath(options.relaxng)
+if options.inputs is not None:
+    inputs_path = os.path.abspath(options.inputs)
+
+# Search for relaxng path if not set
+if options.relaxng is None:
+    xml_validate_path = os.path.abspath(os.path.dirname(sys.argv[0]))
```
Comment (paulromano) [inline src/utils/xml_validate.py]: Shouldn't you be using `args` instead of `sys.argv[0]`?

Comment (bhermanmit) [inline src/utils/xml_validate.py]: Doesn't look like args includes the script name.

Comment (paulromano) [inline src/utils/xml_validate.py]: You're right; sorry.

**Comment by bhermanmit** [comment]
@paulromano - documentation added, let me know if you have any comments. Otherwise, I think it is ready to go pending test suite.

**Comment by bhermanmit** [comment]
I also changed tally bins from string to doubles. After testing a bit, a TypeError was thrown for some of the failures I induced so I added that to the exception handler in the xml_validate script.

**Comment by smharper** [comment]
Also, consider changing the geometry RelaxNG specification for cell rotation from doubles to integers.  I know that we should really be accepting doubles, but we are actually only accepting integers, and I haven't had the time yet to go through and confirm that all our math will work for doubles.

**Comment by paulromano** [comment]
Rotation is definitely intended to be a double. I'd rather you fix how it's read in the input_xml module. Right after the angles (in degrees) are read in, you can see they are converted to radians (stored as doubles).

**Comment by nelsonag** [comment]
I realize I am late on asking this and that is my fault, but is there a reason why we are sticking with rnc files and then relying on an external package (TRANG) to convert rnc to rng? Why don't we just switch to working in RNG in the first place?

**Comment by paulromano** [comment]
I may be the only one, but I use the rnc files for [validation in emacs](http://mit-crpg.github.io/openmc/usersguide/install.html#configuring-input-validation-with-gnu-emacs-nxml-mode). That being said, we can just store rnc and rng files alongside one another so there's no dependence on trang in this script.

**Comment by bhermanmit** [comment]
Works for me, as long as we remember to update both. I'll make the changes to develop in the next couple of days.
________________________________

**Pull request merged**